sql/rowexec/rel.go

```diff
@@ -263,6 +263,12 @@ func (b *BaseBuilder) buildJoinNode(ctx *sql.Context, n *plan.JoinNode, row sql.
 		panic(fmt.Sprintf("%s is a placeholder, RowIter called", n.Op))
 	case n.Op.IsMerge():
 		return newMergeJoinIter(ctx, b, n, row)
+	case n.Op.IsLateralCrossJoin():
```

Comment:
What if there was only one `applyJoinRowIter{filter, left}` rowIter? My intuition is that apply join execution is going to be a nested loop join for all three. There are other similarities - most transform rules are going to treat these all the same right now (because we don't have any apply join optimizations yet), and join reordering ignores all three in the same way, so there are other spots we can maybe deduplicate.